File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Deque<Character> stack = new ArrayDeque<>();
        for (char ch : s.toCharArray()) {
            if (ch == '(' || ch == '{' || ch =='[') stack.push(ch);
            else {
                if (stack.isEmpty()) return false;
                if (ch == ')') {
                    if (stack.peek() != '(') return false;
                    stack.pop();
                }
                else if (ch == '}') {
                    if (stack.peek() != '{') return false;
                    stack.pop();
                }
                else if (ch == ']') {
                    if (stack.peek() != '[') return false;
                    stack.pop();
                }
            }
        }

        return stack.isEmpty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */

class Stack {
    constructor() {
      this.arr = [];
    }

    push(item) {
      this.arr.push(item);
    }

    pop() {
      return this.arr.pop();
    }

    peek() {
        if (this.arr.length === 0) {
            return null;
        }
        return this.arr[this.arr.length - 1];
    }

    getSize() {
        return this.arr.length;
      }

    isEmpty() {
        return this.getSize() === 0;
    }
}

var isValid = function(s) {
    let stack = new Stack();
    
    for (let ch of s.split("")) {
        if (ch === '(' || ch === '{' || ch === '[') {
            stack.push(ch);
        } else {
            if (stack.isEmpty()) return false;
            
            if (ch === ')' && stack.pop() !== '(') return false;
            if (ch === '}' && stack.pop() !== '{') return false;
            if (ch === ']' && stack.pop() !== '[') return false;
        }
    }

    return stack.isEmpty();
};